crates/cast/src/starknet_commands/declare.rs

```diff
@@ -130,29 +130,28 @@ pub async fn declare(
         .iter()
         .find_map(|contract| {
             if contract.contract_name == contract_name {
-                return Some(contract.artifacts.sierra.clone());
+                return contract.artifacts.sierra.clone();
             }
             None
         })
-        .unwrap_or_else(|| {
-            panic!("Failed to find contract {contract_name} in starknet_artifacts.json")
+        .ok_or_else(|| -> anyhow::Result<Utf8PathBuf> {
+            anyhow::bail!("Failed to find contract {contract_name} in starknet_artifacts.json")
```

Comment:
Can't we just 
```suggestion
        .ok_or_else(|| -> {
            anyhow::bail!("Failed to find contract {contract_name} in starknet_artifacts.json")
```
?

Reply:
Ok, so how about `map_err` here? I'm not sure if it applies here but it's shorter 😜

Reply:
as discussed, will leave as is for now (to be improved in https://github.com/foundry-rs/starknet-foundry/issues/352)